### ISSUE TITLE
Feature Request: Compatibility with plotly

### ISSUE THREAD

**Issue by jmgirard** [eclarke/ggbeeswarm#78]
I would love to be able to use plotly to add interactivity to my quasirandom plots. I am assuming that this geom is a special case of point geoms and so it should just require writing a method for the to_basic() generic function in plotly.

https://plotly-r.com/custom-geoms.html

**Comment by eclarke** [comment]
Thanks for the suggestion. I haven't used plotly, but you're correct that the geoms are basically just 'geom_point' with different default 'position' objects. With that in mind, have you tried using plotly with 'position_beeswarm' or 'position_quasirandom'?

**Comment by jmgirard** [comment]
It looks like the issue only arises with plotly when trying to use the custom `text` aesthetic (which lets you customize the tooltip upon hovering a point.

```r
library(ggplot2)
library(ggbeeswarm)
library(plotly)

df <- data.frame(
  x = rnorm(30),
  text = paste0("id: ", 1:30)
)

# As geom
p1 <- ggplot(df, aes(x = x, y = "")) +
  geom_quasirandom()
p1
## looks good
ggplotly(p1)
## looks good


# As position
p2 <- ggplot(df, aes(x = x, y = "")) +
  geom_point(position = position_quasirandom())
p2
## looks good
ggplotly(p2)
## looks good


# As position with text
p3 <- ggplot(df, aes(x = x, y = "", text = text)) +
  geom_point(position = position_quasirandom())
p3
## positioning lost !
ggplotly(p3)
## positioning lost ! >> this is the screenshot below
```

![image](https://user-images.githubusercontent.com/6829033/209571319-5ed9e9d6-dbf2-4821-8ffa-1cd34504aeb6.png)

**Comment by eclarke** [comment]
Interesting, thanks for tracking that down. I'll look into this in the next few weeks (though if you have any interest, pull requests are always welcome!)

**Comment by WatanabeSmith** [comment]
For what it's worth, I managed to have plotly and geom_quasirandom work together in this shiny app with custom hover text (side project on data from a game): https://lysiasthefox.shinyapps.io/swgoh_base_stat_explorer/

However with the 0.7.1 this broke with the removal of `groupOnX` and once I refactor I also found that the `text` aesthetic prevented the scatter as well. Still working towards a fix in the meantime

**Comment by WatanabeSmith** [comment]
I think this is connected to the issue linked here: https://github.com/eclarke/ggbeeswarm/issues/79#issuecomment-1382653523

(uninformed speculation)
I think Plotly is treating each point as it's own group, and making each into their own violin with a dodge.width = 0. Set geom_quasirandom(dodge.width=NULL), that fixed my problems

**Comment by jmgirard** [comment]
Nice job, @WatanabeSmith. This did fix it for me too.

```r
library(ggplot2)
library(ggbeeswarm)
library(plotly)

df <- data.frame(
  x = rnorm(30),
  text = paste0("id: ", 1:30)
)

p3 <- ggplot(df, aes(x = x, y = "", text = text)) +
  geom_point(position = position_quasirandom(dodge.width = NULL))
p3

ggplotly(p3)

# Works!
```

**Comment by eclarke** [comment]
A fix was just merged in #80 that set the default for `dodge.width = NULL`, which should fix this issue. If either of you want to test it out, you should be able to by installing the github version of ggbeeswarm.
